horizon_core/cli_wrapper.py

```diff
@@ -0,0 +1,213 @@
+        ascii_art: Optional[str] = None,
+    ):
+        self.app_name = app_name
+        self.app_description = app_description
+        self.version = version
+        self.ascii_art = ascii_art
+
+        # Initialize rich console
+        self.console = Console()
+
+        # Initialize typer app
+        self.app = typer.Typer(
+            name=app_name, help=app_description, add_completion=True, rich_markup_mode="rich"
+        )
+
+        # Add standard commands
+        self._add_standard_commands()
+
+        # Store interactive commands for help
+        self.interactive_commands: Dict[str, Callable[[], Any]] = {}
+
+    def _add_standard_commands(self):
+        """Add standard CLI commands that all modules should have."""
+
+        @self.app.command()
+        def version():  # type: ignore
+            """Show version information."""
+            self._show_banner()
+            rprint(
+                f"[bold green]{self.app_name}[/bold green]"
+                + f" version [bold blue]{self.version}[/bold blue]"
+            )
+
+        @self.app.command()
+        def interactive():  # type: ignore
+            """Launch interactive mode for guided usage."""
+            self._interactive_mode()
+
+    def _show_banner(self):
+        """Display the application banner with ASCII art."""
+        self.console.clear()
+
+        if self.ascii_art:
+            # Use custom ASCII art
+            banner_text = self.ascii_art
+        else:
+            # Generate ASCII art from app name
+            try:
+                banner_text = figlet_format(self.app_name, font="big")
+            except Exception:
+                banner_text = f"=== {self.app_name.upper()} ==="
+
+        # Create styled banner
+        banner_panel = Panel(
+            Text(banner_text, style="bold cyan"),
+            title=f"[bold green]{self.app_description}[/bold green]",
+            subtitle=f"[dim]v{self.version} - A HorizonSec Module[/dim]",
+            border_style="blue",
+        )
+
+        self.console.print(banner_panel)
+        self.console.print()
+
+    def _interactive_mode(self):
+        """Launch interactive mode with guided menu."""
+        self._show_banner()
+
+        while True:
+            rprint("\n[bold cyan]Interactive Mode[/bold cyan]")
+            rprint("What would you like to do?")
+
+            # Build menu from registered commands
+            choices = ["Show Help", "Exit"]
+            for name in self.interactive_commands.keys():
+                choices.insert(-1, name.title())
+
+            table = Table(show_header=False, box=None)
+            for i, choice in enumerate(choices, 1):
+                table.add_row(f"[bold blue]{i}.[/bold blue]", choice)
+
+            self.console.print(table)
+
+            try:
+                selection = Prompt.ask(
+                    "\nEnter your choice",
+                    choices=[str(i) for i in range(1, len(choices) + 1)],
```

Comment:
From this link https://rich.readthedocs.io/en/latest/prompt.html, why not just enter the choice here instead of creating the table?
```
from rich.prompt import Prompt
name = Prompt.ask("Enter your name", choices=["Paul", "Jessica", "Duncan"], default="Paul")
```